### PR TITLE
Improve the way Json.name is fetched for codegen

### PR DESCRIPTION
**Pull request by ansman**
This depends on https://github.com/square/kotlinpoet/pull/865 being released

### CODE REVIEW

**Comment by ansman** [comment]
Yeah, I suppose it's slightly less gross now

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <okio.version>1.16.0</okio.version>
     <okio2.version>2.1.0</okio2.version>
     <kotlin.version>1.3.60</kotlin.version>
-    <kotlinpoet.version>1.4.4</kotlinpoet.version>
+    <kotlinpoet.version>1.5.0-SNAPSHOT</kotlinpoet.version>
```
Comment (ansman) [inline pom.xml]: Obviously this shouldn't be merged until a stable version is out

**Comment by ansman** [comment]
I'll investigate why it failed. I'll also add a test for json names that contain `"` which currently doesn't work

File: pom.xml [hunk 2]
```diff
@@ -37,7 +37,7 @@
     <okio.version>1.16.0</okio.version>
     <okio2.version>2.1.0</okio2.version>
     <kotlin.version>1.3.60</kotlin.version>
-    <kotlinpoet.version>1.4.4</kotlinpoet.version>
+    <kotlinpoet.version>1.5.0</kotlinpoet.version>
```
Comment (ZacSweers) [inline pom.xml]: I would imagine this brings in a lot of other breaking changes in metadata with the file facade support

**Comment by ZacSweers** [comment]
Thanks!

**Pull request merged**